### PR TITLE
fix: move cancel upgrade plan logic into createSubscriptionWorkflow

### PR DESCRIPTION
**Pull request by BrooksFlannery**
## Summary by cubic
Centralized free-to-paid upgrade handling in createSubscriptionWorkflow to cancel and link the old free subscription during paid plan creation. Added an option to preserve the free plan’s billing cycle when upgrading.

- **Refactors**
  - Moved free-plan cancellation and linking into createSubscriptionWorkflow; processSetupIntent now only forwards preserveBillingCycleAnchor.
  - Removed cancelFreeSubscriptionForUpgrade and previousSubscriptionId usage; upgrade notifications are triggered when a free plan is canceled in workflow.
  - Updated tests to cover billing cycle preservation, event counts, and multi-subscription edge cases.

- **New Features**
  - Added preserveBillingCycleAnchor to carry billingCycleAnchor and current period from the free plan to the new paid subscription, with validation and proration.
  - Guardrails: enforce a single active free subscription per customer; prevent duplicate subscriptions unless the organization allows multiples.

<sup>Written for commit ade7a1eea2eb0661128c3010b2053a7f99dd10a2. Summary will update automatically on new commits.</sup>

### CODE REVIEW

File: platform/flowglad-next/src/subscriptions/createSubscription/workflow.test.ts [hunk 1]
```diff
@@ -1943,3 +1937,136 @@ describe('createSubscriptionWorkflow with discount redemption', async () => {
     })
   })
 })
+
+describe('createSubscriptionWorkflow free plan upgrade behavior', async () => {
+  it('cancels existing free subscription and preserves billing cycle when preserveBillingCycleAnchor is true', async () => {
```
Comment (agreea) [inline platform/flowglad-next/src/subscriptions/createSubscription/workflow.test.ts]: should we test `"resets billing cycle when preserveBillingCycleAnchor is falsey"`?

File: platform/flowglad-next/src/utils/bookkeeping/processSetupIntent.ts [hunk 2]
```diff
@@ -576,46 +513,6 @@ export const createSubscriptionFromSetupIntentableCheckoutSession =
...
-        ),
-        payload: {
-          object: EventNoun.Subscription,
-          id: output.result.subscription.id,
-          customer: {
-            id: customer.id,
-            externalId: customer.externalId,
-          },
-        },
-      })
```
Comment (agreea) [inline platform/flowglad-next/src/utils/bookkeeping/processSetupIntent.ts]: we know this is safe to rm?

File: platform/flowglad-next/src/subscriptions/createSubscription/workflow.test.ts [hunk 3]
```diff
@@ -1943,3 +1937,136 @@ describe('createSubscriptionWorkflow with discount redemption', async () => {
...
+    const { organization, product } = await setupOrg()
+    const customer = await setupCustomer({
+      organizationId: organization.id,
+      livemode: true,
+    })
```
Comment (agreea) [inline platform/flowglad-next/src/subscriptions/createSubscription/workflow.test.ts]: Should we be doing a full `createCustomerBookkeeping` call, and also creating a proper default pricing model with a free plan? So that our test coverage here is as close to reality as possible

File: platform/flowglad-next/src/utils/bookkeeping/processSetupIntent.ts (hunk 2 above)
Comment (BrooksFlannery) [reply platform/flowglad-next/src/utils/bookkeeping/processSetupIntent.ts]: Im pretty sure it is, we create the subscription created event in the workflow as well as line 275 and we also test that we are creating exactly 1 subscription created event in processSetupIntent.upgrade-comprehensive.test.ts at line 1966
